### PR TITLE
feat: support inline asset in `deno_deployment` & rework on `deno_assets`

### LINKED ISSUE #16: Support in-line `content` and `encoding` fields in `deno_deployment`'s `assets` map

### PR DESCRIPTION
**Pull request by magurotuna**
This commit adds support for inlined assets in the `deno_deployment` resource. This also involves some rework on the `deno_assets` data source, which breaks both input and output types.

Now the terraform config that does a deployment using inlined assets looks like as follows:

```terraform
resource "deno_project" "test" {}

resource "deno_deployment" "test" {
  project_id = deno_project.test.id
  entry_point_url = "main.ts"
  assets = {
    "main.ts" = {
      kind = "file"
      content = <<-EOT
        Deno.serve(async () => {
          try {
            const image = await Deno.readFile('computer_screen_programming.png');
            return new Response(image);
          } catch (error) {
            return new Response(error.message);
          }
        });
      EOT
    }
    "computer_screen_programming.png" = {
      kind = "file"
      content = filebase64("testdata/binary/computer_screen_programming.png")
      encoding = "base64"
    }
  }
  env_vars = {}
}
```

Also, terraform's builtin function `merge` can be used to merge normal assets into inlined ones:

```terraform
resource "deno_project" "test" {}

data "deno_assets" "image" {
  path = "testdata/binary"
  pattern = "*.png"
  target = "image"
}

resource "deno_deployment" "test" {
  project_id = deno_project.test.id
  entry_point_url = "main.ts"
  compiler_options = {}
  assets = merge(
    data.deno_assets.image.output,
    {
      "main.ts" = {
        kind = "file"
        content = <<-EOT
          Deno.serve(async () => {
            try {
              const image = await Deno.readFile('image/computer_screen_programming.png');
              return new Response(image);
            } catch (error) {
              return new Response(error.message);
            }
          });
        EOT
      }
    },
  )
  env_vars = {}
}
```

Other examples can be found in `internal/provider/deployment_resource_test.go`.

Resolves #16

### CODE REVIEW

File: docs/data-sources/assets.md [hunk 1]
```diff
@@ -29,18 +30,25 @@ data "deno_assets" "my_assets" {
...
 ### Nested Schema for `output`
 
 Read-Only:
 
-- `git_sha1` (String) The git object hash of the asset. It is only available for `file` asset.
 - `kind` (String) The kind of the asset. It can be either `file` or `symlink`.
-- `target` (String) The target path of the asset. It is only available for `symlink` asset.
-- `updated_at` (String) The last updated time of the asset. It is only available for `file` asset.
+- `local_file_path` (String) The file path of the asset in the local filesystem.
+- `runtime_target_path` (String) The target file path of the symlink in the the runtime virtual filesystem. It is only available for `symlink` asset.
```
Comment (piscisaureus) [inline docs/data-sources/assets.md:54]: Why was this renamed?

File: docs/resources/deployment.md [hunk 2]
```diff
@@ -100,9 +105,10 @@ Required:
 
 Optional:
 
-- `git_sha1` (String) The git object hash for the file. This is valid only for kind == "file".
-- `target` (String) The target file path for the symlink. This is valid only for kind == "symlink".
-- `updated_at` (String) The time the file was last updated. This is valid only for kind == "file".
+- `content` (String) The inlined content of the asset. This is valid only for `file` asset. If both `content` and `local_file_path` are specified, `content` will have higher precedence.
```
Comment (piscisaureus) [inline docs/resources/deployment.md:108]: Rather than specifying that one takes precendence over the other, why not make them mutually exclusive?

File: docs/data-sources/assets.md (hunk 1 above)
Comment (magurotuna) [reply docs/data-sources/assets.md:54]: If it was Rust I would type them distinctly, i.e. `struct LocalPath(String)` and `struct RuntimePath(String)`, where `LocalPath` can be used to get the file content in the local filesystem and `RuntimePath` is set to assets manifest. Just using string type for both of these two path kinds might be very confusing (to me at least), so as a workaround I make them explicit by changing attribute names

File: docs/resources/deployment.md (hunk 2 above)
Comment (magurotuna) [reply docs/resources/deployment.md:108]: We could do that. I'll fix that

Comment (magurotuna) [reply docs/resources/deployment.md]: Fixed and added tests [fb7d120](https://github.com/denoland/terraform-provider-deno/pull/28/commits/fb7d12081e9c41334704930133618060e7d93bca)

**Comment by piscisaureus** [comment]
@magurotuna I made the following changes:
* Symlink target is just `target` to stay consistent with the REST API.
* Local source path for file asset content is now specified with `content_source_path` attribute.
* `encoding` and `content_source_path` are also made mutually exclusive.
Otherwise the PR looks good to me.

If you agree with the above changes, feel free to land, otherwise let's talk about it.

**Review by piscisaureus — APPROVED** [review]
